flow/probes/dpdk.go

```diff
@@ -0,0 +1,242 @@
+// +build dpdk
+
+/*
+ * Copyright (C) 2017 Red Hat, Inc.
+ *
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *  http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ *
+ */
+
+package probes
+
+import (
+	"fmt"
+	"strconv"
+	"sync/atomic"
+
+	"github.com/google/gopacket"
+	"github.com/google/gopacket/layers"
+	dpdkcommon "github.com/intel-go/yanff/common"
+	dpdkflow "github.com/intel-go/yanff/flow"
+	"github.com/intel-go/yanff/packet"
+
+	"github.com/skydive-project/skydive/analyzer"
+	"github.com/skydive-project/skydive/api"
+	"github.com/skydive-project/skydive/config"
+	"github.com/skydive-project/skydive/flow"
+	"github.com/skydive-project/skydive/topology"
+	"github.com/skydive-project/skydive/topology/graph"
+)
+
+var (
+	dpdkNBWorkers uint
+)
+
+// DPDKProbesHandler describes a flow probe handle in the graph
+type DPDKProbesHandler struct {
+	graph          *graph.Graph
+	fta            *flow.TableAllocator
+	flowClientPool *analyzer.FlowClientPool
+}
+
+// RegisterProbe registers a gopacket probe
+func (p *DPDKProbesHandler) RegisterProbe(n *graph.Node, capture *api.Capture, ft *flow.Table, e FlowProbeEventHandler) error {
```

Comment:
You are not using the flow table allocated by the ondemand server. It seems to me that the probe should use the allocator instead of using the flow table allocated. so what I suggest is to pass the allocator to the probes and let them allocate what they need. The allocator should be pass in the constructor

Reply:
By the way you have to use the FlowProbeEventHandler to specify if the capture is started or not same for the unregister